openlibrary/solr/update_work.py

```diff
@@ -1452,7 +1452,7 @@ def solr_escape(query):
     :param str query:
     :rtype: str
     """
-    return re.sub('([\s\-+!()|&{}\[\]^\"~*?:\\\\])', r'\\\1', query)
+    return re.sub(r'([\s\-+!()|&{}\[\]^\"~*?:\\\\])', r'\\\1', query)
```

Comment:
I think this needs to be `\\` instead of `\\\\` after the conversion to raw strings (ie remove the double escaping that was there to work for non-raw strings.

Also, `\"` => `"`

Reply:
> Was a tool used to generate this?

Nope.  :-(  Pretty manual after the following snippet that opens the right files in vscode...

__grep "invalid escape sequence" pytest_output.txt | python3 -c "import sys ; print('code ' + '\n     '.join(sorted(set(x.split()[1].split(':')[0][47:] + ' \\\\' for x in sys.stdin))))"__

The vscode Python plugin's syntax highlighting does help in finding the offending lines but it is a manual process.  Thanks for the review.  I try to steer clear of regex so I am not so good at escape sequences.

Reply:
good catch! Agree, it should be
`return re.sub(r'([\s\-+!()|&{}\[\]^"~*?:\\])', r'\\\1', query)`

This removes the slash infront of the `"` too.

It's trying to replace a range of single characters with an escaped slash followed by the original character. (Took me a while to process the intent!)

Reply:
`\"` => `"` was also done in the last commit.